content/en/blog/_posts/2021-03-09-Defining-NetworkPolicy-Conformance-For-CNIs

```diff
@@ -0,0 +1,373 @@
+
+The good news is, all of these issues are actively being fixed and iterated on between sig-network, the CNI providers, and the network policy subproject.  
+
+If you're interested in verifying NetworkPolicy functionality on your cluster (and, if you care about security, or run a SaaS, you should be), then you can run the upstream end to end tests, or cyclonus, or both.
+- If you are deeply curious about your CNI providers network policy implementation, use cyclonus to test *100s* of policies, and evaluate your CNI for comprehensive functionality, for deep discovery of potential security holes, or for getting involved with the upstream networkpolicy efforts.
+- If your just getting started with network policies, your better off running the e2e tests if you want to simply verify the "common" network policy cases that most CNIs should be implementing correctly, in a way that is quick to diagnose.
+
+## Where to start with NetworkPolicy testing ?
+
+- You can run cyclonus on your cluster easily by following the instructions at https://github.com/mattfenwick/cyclonus, and determine wether *your* specific CNI configuration is
+fully conformant to the 100s of different Kubernetes Network Policy API constructs.
+- Alternatively, you can use a tool like `sonobuoy` (https://github.com/vmware-tanzu/sonobuoy) to run the existing E2E tests in Kubernetes, with the `--ginkgo.focus=NetworkPolicy` flag.  Note that only versions of sonobuoy built for k8s 1.21 and above will have the *new* network policy tests in them.
+
+## Improvements to the NetworkPolicy API and User Experience
+
+In addition to cleaning up the validation story for CNI providers over NetworkPolicies, we've also spent some time improving the K8s API for a few commonly requested NetworkPolicy features.
+After months of delibration, we eventually settled on a few core areas for improvement:
+
+- Port Range policies: We now allow you to specify a *range* of ports for a policy.  This allows users interested in scenarios like FTP or virtualization to enable advanced policies.  The port range
+option for networkpolicies will be available to use in Kubernetes 1.21.  You can read more about this here https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2079-network-policy-port-range. 
+- Namespace as name policies: Allowing users in Kubernetes 1.21 and forward to target namespaces by default labels that are always on all namespaces.  This was done in collaboration with jordan ligget and tim hockins on the apimachinery side.
+This change allowed us to improve the network policy user experience without actually changing the API ! For details, you can read about it here https://github.com/kubernetes/enhancements/pull/2162. 
+The TLDR is that after Kubernetes 1.21, ALL NAMESPACES will have the following label, addd by default: 
+
+```
+kubernetes.io/metadata.name: my-namespace
+```
```

Comment:
Is this a firm commitment? If not, I recommend rewording to phrase it clearly as an ambition.
Also “addd” is a typo.

Reply:
I believe I have addressed that @jayunit100 @rikatz please review.